src/MSBuild/XMake.cs

```diff
@@ -1725,6 +1725,18 @@ internal static void GatherCommandLineSwitches(ArrayList commandLineArgs, Comman
                                 String.Equals(switchName, "maxcpucount", StringComparison.OrdinalIgnoreCase))
                             {
                                 int numberOfCpus = Environment.ProcessorCount;
+#if !MONO
+                                // .NET framework calls Windows APIs that have a core count limit (32/64 depending on process bitness).
+                                // .NET Core on Windows returns a core count limited to the current NUMA node
+                                //     https://github.com/dotnet/runtime/issues/41902
+                                // So if we get a high core count on Windows, double-check it.
+                                if (numberOfCpus >= 32 && NativeMethodsShared.IsWindows)
```

Comment:
Were we not thinking of expanding this for cases like, say, if you have 2 * 24 CPUs?

Reply:
Ah, indeed we were. Back to the drawing board.

Reply:
Ok, latest push looks good again.